src/server/middleware/blob/patch_blob_upload.go

```diff
@@ -39,7 +40,13 @@ func PatchBlobUploadMiddleware() func(http.Handler) http.Handler {
 
 		sessionID := distribution.ParseSessionID(r.URL.Path)
 
-		return blobController.SetAcceptedBlobSize(r.Context(), sessionID, size)
+		// Only cache non-zero sizes in Redis
+		if size > 0 {
```

Comment:
You are checking here for >0 zero and in SetAcceptedBlobSize. 
A change here isn't needed